application/src/main/java/org/opentripplanner/graph_builder/module/osm/edgelevelinfo/InclinedEdgeLevelInfoProcessor.java

```diff
@@ -0,0 +1,26 @@
+package org.opentripplanner.graph_builder.module.osm.edgelevelinfo;
+
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.opentripplanner.graph_builder.module.osm.OsmDatabase;
+import org.opentripplanner.osm.model.OsmWay;
+import org.opentripplanner.service.streetdetails.StreetDetailsRepository;
+import org.opentripplanner.service.streetdetails.model.InclinedEdgeLevelInfo;
+import org.opentripplanner.street.model.edge.Edge;
+
+/**
+ * Contains logic for storing edge level info in the
+ * {@link StreetDetailsRepository}.
+ */
+public interface InclinedEdgeLevelInfoProcessor {
+  InclinedEdgeLevelInfoProcessor NOOP = new NoopInclinedEdgeLevelInfoProcessor();
+
+  public Optional<InclinedEdgeLevelInfo> findInclinedEdgeLevelInfo(OsmDatabase osmdb, OsmWay way);
+
+  public void storeLevelInfoForEdge(
+    @Nullable Edge forwardEdge,
+    @Nullable Edge backwardEdge,
+    Optional<InclinedEdgeLevelInfo> inclinedEdgeLevelInfoOptional,
```

Comment:
I don't understand why but the Java world has a strong aversion to passing Optional as a method parameter :shrug: . So lets respect local culture and use a nullable value.

Reply:
I changed this parameter to a non-nullable value and I moved the empty check into osmmodule